paddle/scripts/docker/Dockerfile.m4

```diff
@@ -1,12 +1,36 @@
 FROM PADDLE_BASE_IMAGE
 MAINTAINER PaddlePaddle Dev Team <paddle-dev@baidu.com>
-COPY build.sh /root/
-ENV GIT_CHECKOUT=v0.9.0a0
+
+# It is good to run apt-get install with Dockerfile RUN directive,
+# because if the following invocation to /root/build.sh fails, `docker
+# build` wouldn't have to re-install packages after we fix
+# /root/build.sh.
+RUN apt-get update && \
+    apt-get install -y cmake libprotobuf-dev protobuf-compiler git \
+    libgoogle-glog-dev libgflags-dev libatlas-dev libatlas3-base g++ m4 python-pip \
+    python-protobuf python-numpy python-dev swig openssh-server \
+    wget unzip python-matplotlib tar xz-utils bzip2 gzip coreutils \
+    sed grep graphviz libjpeg-dev zlib1g-dev doxygen && \
+    apt-get clean -y
+RUN pip install BeautifulSoup docopt PyYAML pillow \
+    'sphinx>=1.4.0' sphinx_rtd_theme breathe recommonmark
+
 ENV WITH_GPU=PADDLE_WITH_GPU
-ENV IS_DEVEL=PADDLE_IS_DEVEL
-ENV WITH_DEMO=PADDLE_WITH_DEMO
-ENV PIP_INSTALL_ARGS ""
-ENV PIP_GENERAL_ARGS ""
-ENV USE_UBUNTU_MIRROR OFF
 ENV WITH_AVX=PADDLE_WITH_AVX
-RUN cd /root/ && bash build.sh
+
+RUN mkdir /paddle
+COPY . /paddle/
+COPY paddle/scripts/docker/build.sh /root/
+RUN /root/build.sh
+
+RUN echo 'export LD_LIBRARY_PATH=/usr/lib64:${LD_LIBRARY_PATH}' >> /etc/profile
+RUN pip install /usr/local/opt/paddle/share/wheels/*.whl
+RUN paddle version  # print version after build
+
+# Configure OpenSSH server. c.f. https://docs.docker.com/engine/examples/running_ssh_service
```

Comment:
I can roughly understand why there is need for openssh server (e,g., ssh in and copy files out). That could be achieved by mouting a local folder (`-v`). Just want to understand the use cases more, are there any discussion document about the reasoning behind packaging in a ssh server?